activerecord/lib/active_record/relation/batches.rb

```diff
@@ -219,5 +221,19 @@ def apply_limits(relation, start, finish)
     def batch_order
       "#{quoted_table_name}.#{quoted_primary_key} ASC"
     end
+
+    def order_or_limit_ignored_message
+      "Scoped order and limit are ignored, it's forced to be batch order and batch size".freeze
+    end
+
+    def act_on_order_or_limit_ignored(error_on_ignore)
+      error_on_ignored_order_or_limit = (error_on_ignore.nil? ? ActiveRecord::Base.error_on_ignored_order_or_limit : error_on_ignore)
```

Comment:
`self.klass.error_on_ignored_order_or_limit` to make sure this config can be set per model